### ISSUE TITLE
add conditions into Nix Language Basics tutorial

### ISSUE THREAD

**Issue by ellyxir** [NixOS/nix.dev#1079]
**Observations**
After finishing the language basics, I thought about writing some basic code and realised that I didn't know how to use any conditionals. I thought maybe it was in the builtins but didn't see it there. Searching showed it was `if` in the Nix ref manual under Language constructs. Perhaps I missed it in the tutorial?

**Problem**
It feels like conditionals are a basic part of the language that I'd expect to know from a first pass tutorial. In fact, the tutorial touches on all language constructs except for conditionals and asserts. Perhaps comments also.

**Approaches**
Add conditionals to the language basics tutorial.

**Willing to help?**
I'm willing to help.

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix.dev/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by fricklerhandwerk** [comment]
Yes, I originally made the decision to skip conditionals because there's really not much to say about them for the purpose of being able to read the language. In real-world code you rarely find `if ... then ... else` and when you do they're unsurprising. The purpose of the tutorial is to prepare people for what's unusual and special about the Nix language, and that taking enough time and space as it is made me conclude it's better to skip the parts that will be obvious for people experienced with programming.

But since it's been brought up a couple of times now, I can imagine that maybe we mention it but still not go into detail to keep the noise down. What do you think about #1082?

**Comment by fricklerhandwerk** [comment]
btw, thanks a lot @ellyxir for bringing up these issues and making improvements! A lot of learners and new users will benefit from those, I'm sure. :)

**Comment by bew** [comment]
What about having a section saying that conditionals are unsurprising, to at least mention it and give a simple example 🤔

Something that could be surprising tho' coming from imperative language is that the `else ...` branch is required.

**Comment by fricklerhandwerk** [comment]
> Something that could be surprising tho' coming from imperative language is that the `else ...` branch is required.

Yeah that's true and probably worth mentioning. I'm still not sure if it's worth an entire section though, and if it is, where to put it. Feels weird to squeeze it somewhere randomly in between.

**Comment by ellyxir** [comment]
> But since it's been brought up a couple of times now, I can imagine that maybe we mention it but still not go into detail to keep the noise down. What do you think about #1082?

That would have worked, I like that its near the beginning because I did a search for 'if' and it would have shown up early, thank you!

**Comment by fricklerhandwerk** [comment]
Thanks for the feedback!
